animation_nodes/nodes/mesh/set_bevel_edge_weight.py

```diff
@@ -0,0 +1,31 @@
+import bpy
+from ... data_structures import VirtualDoubleList
+from ... base_types import AnimationNode, VectorizedSocket
+
+class BevelEdgeWeight(bpy.types.Node, AnimationNode):
+    bl_idname = "an_SetBevelEdgeWeight"
+    bl_label = "Set Bevel Edge Weight"
+    errorHandlingType = "EXCEPTION"
+
+    useWeightList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput("Object", "Object", "object", defaultDrawType = "PROPERTY_ONLY")
+        self.newInput(VectorizedSocket("Float", "useWeightList",
+            ("Weight", "weight"), ("Weights", "weights")))
+
+        self.newOutput("Object", "Object", "object")
+
+    def execute(self, object, weights):
+        if object is None: return None
+
+        if object.type != "MESH" or object.mode != "OBJECT":
+            self.raiseErrorMessage("Object is not in object mode or is no mesh object.")
+
+        if object.data.use_customdata_edge_bevel is False:
```

Comment:
Is there a good reason why you are checking if it is false first?

Also, the if statement should be `if not object.data.use_customdata_edge_bevel`, if it is actually needed.

Reply:
If `object.data.use_customdata_edge_bevel` is already true then we need no to set it again. 

However, we can also use this code only `object.data.use_customdata_edge_bevel = True`.

Reply:
If it does not trigger any recalculations, then I prefer if you do it that way.

Reply:
With the if-condtion is faster, it took Min-Time `0.0258ms` less as compared to without the if-condition.

Reply:
This seems negligible, but sure, keep the if.